pkg/controller/clusterissuers/sync.go

```diff
@@ -41,8 +41,8 @@ const (
 func (c *controller) Sync(ctx context.Context, iss *cmapi.ClusterIssuer) (err error) {
 	log := logf.FromContext(ctx)
 
-	// allow a maximum of 10s
-	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
+	// allow a maximum of 90s
+	ctx, cancel := context.WithTimeout(ctx, time.Second*90)
```

Comment:
First of all thank you for doing this! :heart: 

question: You say 90s would solve most of your problems, but some people in the team feel like 90s is a long time. We don't really have any data on this but and once we increase the timeout we can't easily decrease it since people might view that as a backwards-incompatible change.

I think [you're the person](https://github.com/cert-manager/cert-manager/issues/5080#issuecomment-1137259557) whose mentioned the longest request durations I've seen. Do you have numbers on how long various operations such as issuer creation and certificate issuance take to complete?

In short: We want this timeout to be as low as possible but we're desperate for more numbers on how low we can go. Could we go to 60s?

note: It's also worth pointing out that this change _won't_ affect certificate issuance timeouts. That might be fine for this PR to keep this PR small, but it's worth noting.

Reply:
@SgtCoDFish may be give the user possibility to control the timeout through environment variables and then everyone can set the timeout for themselves?

Reply:
> In short: We want this timeout to be as low as possible but we're desperate for more numbers on how low we can go. Could we go to 60s?

I would say 60s would be feasible, high chance of it going through with 60s than 10s. So if Sectigo were to be slow at the first setup of cluster issuer, it might be just fast enough the second try. So instead of it taking 8 hours to create the cluster issuer, it might take 30 minutes if Sectigo's API decides to be slow.

Then later, as @SgtCoDFish says, cert-manager gets a flag for configuring timeouts, so if 60s is deemed too slow, one can adjust it oneself, and still keep the default at 60s.

Reply:
For context, certbot sets a timeout of 45s per HTTP request, and 90s per ACME order: https://github.com/certbot/certbot/blob/295fc5e33a68c945d2f62e84ed8e6aaecfe93102/acme/acme/client.py

Reply:
@MilheiroSantos Very useful addition, I think it would be a good strategy to follow Certbots decisions here. Should I also increase the timeout for issuing to 45 seconds @SgtCoDFish?

Reply:
I think adjusting issuance, as @SgtCoDFish has mentioned, would be its own PR to not get too much scope creep, and not adjusting too much in this PR.

Reply:
If we see that adjusting (Cluster)Issuer deadline works fine, I'd presume a new PR with adjusting issuance timeout would get through smoother.